extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -151,65 +156,48 @@ public void build(ApplicationInfoBuildItem applicationInfo,
                 .map(KubernetesDeploymentTargetBuildItem::getName)
                 .collect(Collectors.toSet());
 
-        // this is a hack to get kubernetes.registry working because currently it's not supported as is in Dekorate
-        Optional<String> dockerRegistry = KubernetesConfigUtil.getDockerRegistry(config);
-        dockerRegistry.ifPresent(v -> System.setProperty(DOCKER_REGISTRY_PROPERTY, v));
-
-        // this is a hack to work around Dekorate using the default group for some of the properties
-        Optional<String> kubernetesGroup = KubernetesConfigUtil.getGroup(config);
-        kubernetesGroup.ifPresent(v -> System.setProperty(APP_GROUP_PROPERTY, v));
```

Comment:
With these changes `DOCKER_REGISTRY_PROPERTY` and `APP_GROUP_PROPERTY` are no longer used